File: Java/Program/Clothing.java

public class Clothing extends Product {
    private String size;
    private String material;
    private String gender;

    public Clothing(String name, String brand, int price, String size, String material, String gender) {
        super(name, brand, price);
        this.size = size;
        this.material = material;
        this.gender = gender;
    }

    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public String getMaterial() {
        return material;
    }

    public void setMaterial(String material) {
        this.material = material;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }
}

File: CPP/Program/Clothing.cpp
#include <iostream>
#include <string>
#include "Product.h"

class Clothing : public Product {
private:
    std::string size;
    std::string material;
    std::string gender;

public:
    Clothing(std::string name, std::string brand, int price, std::string size, std::string material, std::string gender)
        : Product(name, brand, price) {
        this->size = size;
        this->material = material;
        this->gender = gender;
    }

    std::string getSize() {
        return size;
    }

    void setSize(std::string size) {
        this->size = size;
    }

    std::string getMaterial() {
        return material;
    }

    void setMaterial(std::string material) {
        this->material = material;
    }

    std::string getGender() {
        return gender;
    }

    void setGender(std::string gender) {
        this->gender = gender;
    }
};
